src/utils.rs

```diff
@@ -670,6 +670,52 @@ impl Drop for FlushToZeroDenormalsAreZeroFlags {
     }
 }
 
+// This is an RAII structure that disables floating point exceptions while
+// it is alive, so that operations which generate NaNs and infinite values
+// intentionally will not trip an exception when debugging problematic
+// code that is generating NaNs and infinite values erroneously.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub(crate) struct DisableFloatingPointExceptionsFlags {
+    #[cfg(feature = "avoid-fe-exceptions")]
+    // We can't get a precise size for this, because it's of type
+    // `fenv_t`, which is a definition that doesn't exist in rust
+    // (not even in the libc crate, as of the time of writing.)
+    // But since the state is intended to be stored on the stack,
+    // 256 bytes should be more than enough.
+    original_flags: [u8; 256],
+}
+
+#[cfg(feature = "avoid-fe-exceptions")]
+extern "C" {
+    fn feholdexcept(env: *mut std::ffi::c_void);
+    fn fesetenv(env: *const std::ffi::c_void);
```

Comment:
Technically, these functions will only exist on Linux (maybe Mac? I don't know!) so only checking for the feature flag is insufficient here. However, the windows version of this code would be completely different, and I haven't really got a good windows development environment to figure that out on. However, thinking about it, I think the way it is now is actually good -- simply compiling it out if the flag is set on windows might silently confuse users. If the linker fails when the feature flag is enabled on windows, that's a good sign not to use it yet, because it isn't yet implemented on windows.

But maybe there's a better way to just disable the flag on windows. I'm new to feature flags!